modules/oe_multilingual_url_suffix/src/Plugin/LanguageNegotiation/LanguageNegotiationUrlSuffix.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation;
+
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\language\Plugin\LanguageNegotiation\LanguageNegotiationUrl;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class LanguageNegotiationUrlSuffix for identifying language via URL suffix.
+ *
+ * @LanguageNegotiation(
+ *   id = \Drupal\oe_multilingual_url_suffix\Plugin\LanguageNegotiation\LanguageNegotiationUrlSuffix::METHOD_ID,
+ *   types = {\Drupal\Core\Language\LanguageInterface::TYPE_INTERFACE,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_CONTENT,
+ *   \Drupal\Core\Language\LanguageInterface::TYPE_URL},
+ *   weight = -10,
+ *   name = @Translation("URL suffix"),
+ *   description = @Translation("Language from the URL (Path suffix)."),
+ *   config_route_name = "oe_multilingual_url_suffix.negotiation_url_suffix"
+ * )
+ */
+class LanguageNegotiationUrlSuffix extends LanguageNegotiationUrl {
+
+  /**
+   * The language negotiation method id.
+   */
+  const METHOD_ID = 'language-url-suffix';
+
+  /**
+   * The suffix delimiter.
+   */
+  const SUFFIX_DELIMITER = '_';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLangcode(Request $request = NULL) {
+    $langcode = NULL;
+
+    $config = $this->config->get('oe_multilingual_url_suffix.settings')->get('url_suffixes');
+    if ($request && $config) {
+      $request_path = urldecode(trim($request->getPathInfo(), '/'));
+      $parts = explode(static::SUFFIX_DELIMITER, $request_path);
+      $suffix = array_pop($parts);
+
+      // Search suffix within configs.
+      $negotiated_language = array_search($suffix, $config);
+      if ($negotiated_language) {
+        $langcode = $negotiated_language;
+      }
+    }
+
+    return $langcode;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request) {
+    $config = $this->config->get('oe_multilingual_url_suffix.settings')->get('url_suffixes');
```

Comment:
If you want to check if the config exists you should do it on `$this->config->get('oe_multilingual_url_suffix.settings')`, the next `->get` it's too late. Also i'm not sure it's necessary, this configuration is staged by the module so we know it's there, but other reviewers might have arguments in favor of an extra checking.

Reply:
This is how work \Drupal\Core\Config\ConfigFactory::doGet. It creates config object on fly.
So checking $this->config->get('oe_multilingual_url_suffix.settings') - will not work.
But I have applied some changes in code.